### PR TITLE
[Breaking changes] Array, Nested, Array of Nested, Nested of Nested

### PR DESCRIPTION
**Pull request by simkim**

### CODE REVIEW

File: lib/couchbase-orm/types/array.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+            def cast(values)
+              return [] if values.nil?
+              raise ArgumentError, "#{values.inspect} must be an array" unless values.is_a?(::Array)
+              values.map(&@type_class.method(:cast))
+            end
+        
+            def serialize(values)
+                return [] if values.nil?
+                values.map(&@type_class.method(:serialize))
```
Comment (pimpin) [inline lib/couchbase-orm/types/array.rb:25]:
```suggestion
                return [] if values.nil?

                values.map(&@type_class.method(:serialize))
```

File: lib/couchbase-orm/types/array.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                else
+                    @type_class = ActiveModel::Type.registry.lookup(type)
+                end
+                super()
+            end
+
+            def cast(values)
+              return [] if values.nil?
+              raise ArgumentError, "#{values.inspect} must be an array" unless values.is_a?(::Array)
+              values.map(&@type_class.method(:cast))
```
Comment (pimpin) [inline lib/couchbase-orm/types/array.rb:20]:
```suggestion
            def cast(values)
              return [] if values.nil?
              raise ArgumentError, "#{values.inspect} must be an array" unless values.is_a?(::Array)
              
              values.map(&@type_class.method(:cast))
```

File: lib/couchbase-orm/types/nested.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+module CouchbaseOrm
+    module Types
+        class Nested < ActiveModel::Type::Value
+            attr_reader :model_class
+
+            def initialize(type:)
+                raise ArgumentError, "type is nil" if type.nil?
+                raise ArgumentError, "type is not a class : #{type.inspect}" unless type.is_a?(Class)
+                @model_class = type
```
Comment (pimpin) [inline lib/couchbase-orm/types/nested.rb:9]:
```suggestion
                raise ArgumentError, "type is not a class : #{type.inspect}" unless type.is_a?(Class)

                @model_class = type
```

**Review by pimpin — COMMENTED** [review]
Some cosmetics suggestions on empty lines after early raise or return.

**Review by fab-girard — CHANGES_REQUESTED** [review]
Can you add 2 sections for array and nested attributes with examples in readme please?

**Comment by avsej** [comment]
when this PR will be merged, I will rebase my #55 and fix conflicts if there will be detected.

**Review by avsej — APPROVED** [review]

**Pull request merged**